code/controllers/subsystems/documents.dm

```diff
@@ -0,0 +1,231 @@
+		load_from_sql()
+	else if(config.docs_load_docs_from == "json")
+		log_debug("SSdocs: Attempting to Load from JSON")
+		load_from_json()
+	else
+		log_game("SSdocs: invalid load option specified in config")
+
+	..()
+
+/datum/controller/subsystem/docs/New()
+	NEW_SS_GLOBAL(SSdocs)
+
+/*
+	Fetching Data
+*/
+//A rewritten version of pickweight.
+/datum/controller/subsystem/docs/proc/pick_document()
+	var/subtotal = rand() * src.total_docs
+	for (var/doc in docs)
+		subtotal -= docs[doc].chance
+		if (subtotal <= 0)
+			return docs[doc]
+	return null
+
+//Pick a document by one tag
+/datum/controller/subsystem/docs/proc/pick_document_by_tag(var/tag)
+	if(!docs_by_tags[tag])
+		return null
+	var/subtotal = rand() * src.total_by_tags[tag]
+	for (var/doc in docs_by_tags[tag])
+		subtotal -= docs_by_tags[tag][doc].chance
+		if (subtotal <= 0)
+			return docs_by_tags[tag][doc]
+	return null
+
+//Pick a document by any tag from a list of tags. Weighted.
+/datum/controller/subsystem/docs/proc/pick_document_by_any_tag(var/list/tags)
+	if(!istype(tags) || !tags.len)
+		return null
+	var/total_chance = 0
+	var/tag_sublist = list()
+	for(var/t in tags)
+		if(!total_by_tags[t])
+			return null
+		total_chance += total_by_tags[t]
+		tag_sublist += docs_by_tags[t]
+	var/subtotal = total_chance * rand()
+	for(var/doc in tag_sublist)
+		subtotal -= tag_sublist[doc].chance
+		if(subtotal <= 0)
+			return tag_sublist[doc]
+	return null
+
+//Pick a document by multiple tags that it must have.
+/datum/controller/subsystem/docs/proc/pick_document_by_tags(var/list/tags)
+	if(!istype(tags) || !tags.len)
+		return null
+	var/list/tag_sublist = docs_by_tags[pick(tags)] // the list cannot start off as empty
+	for(var/t in tags)
+		if(!docs_by_tags[t])
+			return null
+		tag_sublist &= docs_by_tags[t]
+	log_ss("docs", "Tag sublist has length [tag_sublist.len].")
+	var/subtotal = 0
+	for(var/datum/docs_document/dd in tag_sublist)
+		subtotal += dd.chance
+	subtotal *= rand()
+	for (var/doc in tag_sublist)
+		subtotal -= tag_sublist[doc].chance
+		if (subtotal <= 0)
+			return tag_sublist[doc]
+	return null
+/*
+	Loading Data
+*/
+//Reset docs to prep for loading in new items
+/datum/controller/subsystem/docs/proc/reset_docs()
+	docs = list()
+	docs_by_tags = list()
+	total_docs = 0
+	total_by_tags = list()
+
+//Load the document data from SQL
+/datum/controller/subsystem/docs/proc/load_from_sql()
+	if(!establish_db_connection(dbcon))
+		log_debug("SSdocs: SQL ERROR - Failed to connect. - Falling back to JSON")
+		return load_from_json()
+	else
+		//Reset the currently loaded data
+		reset_docs()
+
+		//Load the categories
+		var/DBQuery/document_query = dbcon.NewQuery("SELECT name, title, chance, content, tags FROM ss13_documents WHERE deleted_at IS NULL")
+		document_query.Execute()
+		while(document_query.NextRow())
+			CHECK_TICK
+			try
+				add_document(
+					document_query.item[1],
+					document_query.item[2],
+					document_query.item[3],
```

Comment:
You either need to either convert the passed parameter here to an int

Reply:
```suggestion
					text2num(document_query.item[3]),
```